### DISCUSSION TITLE
Can I keep a thread alive on android?

### DISCUSSION THREAD

**Discussion by UnbelievableWang** [beeware/briefcase#637] [Q&A]
I need to keep a thread alive after I press a button,but the thread will be killed on android.
It works well on windows.
 ### My code:
```
"""
My first application
"""
import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW
from toga.widgets.multilinetextinput import MultilineTextInput
from travertino.constants import LEFT
from .many import killer
import threading   
import time

        

def pr():
    while(True):
        print("to check if the thread is working ")

class ww(toga.App):
    def startup(self):
        label=toga.Label("massage")
        mul=toga.MultilineTextInput(readonly=True)
        input = toga.TextInput(placeholder='Information',readonly=True)
        
        path_app=str(self.paths.app)
        print(path_app)


        input.enabled=False
        def connect(bt):
            print(bt)
            label.text="running"
            threading.Thread(target=pr,args=()).start()
            bt.enabled=False
            mul.value="Information will be displayed"
            #time.sleep(1999)
        main_box = toga.Box()
        
        button=toga.Button('run',on_press=connect)
        main_box.add(button)
        main_box.add(label)
        main_box.add(input)
        subbox=toga.Box()
        subbox.add(mul)
        main_box.add(subbox)
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()


def main():
    return ww()
```

I used `adb logcat -s MainActivity:* stdio:* Python:*` to check if the thread is working.
When I enabled `time.sleep(1999)`, it prints logs .But  the whole app is enabled.

**Comment by freakboy3742** [comment]
My suspicion is that the thread object is being garbage collected because it is passing out of scope when the handler completes. If you maintain a reference to the thread object outside the `connect()` handler, the thread should continue to exist, and will co-exist with the main application thread.

**Reply by UnbelievableWang** [reply]
I have tried moving it out of connect to main() or even `__main__.py.`  It also can’t keep alive printing the message. I turned on adb to observe the situation. After I click the button, there will be nothing else printing in my console.

**Reply by UnbelievableWang** [reply]
Only when I run it in emulator,It happens. When I
use the dev mode, It can keep printing things as I expect.
